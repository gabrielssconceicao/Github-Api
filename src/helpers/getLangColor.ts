export const languageColors: Record<string, string> = {
  JavaScript: '#f7df1e',
  Python: '#3572A5',
  Java: '#b07219',
  'C++': '#f34b7d',
  C: '#555555',
  'C#': '#178600',
  PHP: '#4F5D95',
  TypeScript: '#2b7489',
  Swift: '#ffac45',
  Ruby: '#CC342D',
  Go: '#00ADD8',
  R: '#198CE7',
  Kotlin: '#F18E33',
  Scala: '#c22d40',
  Rust: '#dea584',
  Dart: '#00B4AB',
  Perl: '#0298c3',
  Lua: '#000080',
  Matlab: '#e16737',
  Haskell: '#5e5086',
  SQL: '#e38c00',
  HTML: '#e34c26',
  CSS: '#563d7c',
  'Objective-C': '#438eff',
  Assembly: '#6E4C13',
  Shell: '#89e051',
  PowerShell: '#012456',
  Groovy: '#e69f56',
  Vue: '#41B883',
  Dockerfile: '#384d54',
  Clojure: '#db5855',
  Elixir: '#6e4a7e',
  D: '#ba595e',
  Julia: '#a270ba',
  Arduino: '#bd79d1',
  Verilog: '#b2b7f8',
  VHDL: '#adb2cb',
  Ada: '#02f88c',
  Scheme: '#1e4aec',
  'Objective-C++': '#6866fb',
  Crystal: '#000100',
};
export const getLangColor = (lang: string = ''): string => {
  let language = '#ffffff';
  if (lang in languageColors) {
    language = languageColors[lang];
  }
  return language;
};
