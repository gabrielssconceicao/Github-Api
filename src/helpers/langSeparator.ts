import type { IRepos } from '@/types/repos';
type ILang = { name: string; count: number };
export const langSeparator = (repos: IRepos[]): ILang[] => {
  const langs: ILang[] = [];
  repos.forEach((repo) => {
    const lang = repo.language;
    const index = langs.findIndex((l) => l.name === lang);
    console.log(index);
    if (index === -1) {
      langs.push({ name: lang, count: 1 });
    } else {
      langs[index].count += 1;
    }
    console.log(langs);
  });
  return langs;
};
