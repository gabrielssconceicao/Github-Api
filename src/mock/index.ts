import { type IUser } from '@/types/user';
import { type IRepos } from '@/types/repos';

export const user: IUser = {
  name: 'Teste User',
  login: 'testeuser',
  avatar_url: 'https://avatars.githubusercontent.com/u/10590076?v=4',
  followers: 0,
  following: 5,
  location: 'Brazil',
  html_url: 'https://github.com/testeuser',
  bio: 'lorem ipsum dolor sit amet',
  public_repos: 5,
};
// JS - 3
// HTML - 2
// python - 1
// Ruby - 1
//Java - 1
//C++ - 1
//Typescript - 1
export const repos: IRepos[] = [
  {
    name: 'Teste Repo 1',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 2',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 3',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Python',
  },
  {
    name: 'Teste Repo 4',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 5',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 6',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 7',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Ruby',
  },
  {
    name: 'Teste Repo 8',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Java',
  },
  {
    name: 'Teste Repo 9',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'C++',
  },
  {
    name: 'Teste Repo 10',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Typescript',
  },
];
