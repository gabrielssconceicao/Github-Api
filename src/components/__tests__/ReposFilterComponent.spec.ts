import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import ReposFilter from '../ReposFilterComponent.vue';

import { repos } from '@/mock';
import { langSeparator } from '@/helpers/langSeparator';

const separedLangs = langSeparator(repos);

describe('<ReposFilter', () => {
  it('should render correctly', () => {
    const wrapper = mount(ReposFilter);
    expect(wrapper.exists()).toBe(true);
    const btns = wrapper.findAll('button');
    expect(btns.length).toBe(1);
    expect(btns[0].text()).toBe('Clear');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with filter', () => {
    const wrapper = mount(ReposFilter, {
      props: {
        filters: separedLangs,
      },
    });

    const btns = wrapper.findAll('button');
    expect(btns.length).toBe(separedLangs.length + 1);

    for (let i = 1; i < btns.length; i++) {
      expect(btns[i].text()).toBe(
        `${separedLangs[i - 1].name}${separedLangs[i - 1].count}`,
      );
    }

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should check if when the button is clicked', async () => {
    const wrapper = mount(ReposFilter, {
      props: {
        filters: separedLangs,
      },
    });

    const btns = wrapper.findAll('button');

    await btns[2].trigger('click');

    expect(wrapper.emitted('filter')).toBeTruthy();
    expect(wrapper.emitted('filter')).toHaveLength(1);
    expect(btns[2].classes()).includes('selected');

    await btns[0].trigger('click');
    expect(btns[2].classes()).not.includes('selected');

    await btns[2].trigger('click');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should have the name No Language', () => {
    const reposWithNull = [...repos];
    reposWithNull[1].language = null;
    const wrapper = mount(ReposFilter, {
      props: {
        filters: langSeparator(reposWithNull),
      },
    });

    const btns = wrapper.findAll('button');

    expect(btns[2].text()).toBe('No Language1');
    expect(wrapper.element).toMatchSnapshot();
  });
});
