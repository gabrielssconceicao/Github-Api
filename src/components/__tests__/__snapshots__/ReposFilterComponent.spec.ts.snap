// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ReposFilter /> > should have class selected when clicked and unselected when click in another button 1`] = `
<div
  class="w-full grid grid-cols-2 gap-2 md:flex md:justify-around md:flex-wrap p-2"
  data-v-b209dafd=""
>
  
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #ffffff; --hover-text-color: #111111;"
    value="Javascript"
  >
    <span
      data-v-b209dafd=""
    >
      Javascript
    </span>
    <span
      data-v-b209dafd=""
    >
      3
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(227, 76, 38); --hover-bg-color: #e34c26; --hover-text-color: #111111;"
    value="HTML"
  >
    <span
      data-v-b209dafd=""
    >
      HTML
    </span>
    <span
      data-v-b209dafd=""
    >
      2
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(53, 114, 165); --hover-bg-color: #3572A5; --hover-text-color: #111111;"
    value="Python"
  >
    <span
      data-v-b209dafd=""
    >
      Python
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class="selected"
    data-v-b209dafd=""
    style="color: rgb(204, 52, 45); --hover-bg-color: #CC342D; --hover-text-color: #111111;"
    value="Ruby"
  >
    <span
      data-v-b209dafd=""
    >
      Ruby
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(176, 114, 25); --hover-bg-color: #b07219; --hover-text-color: #111111;"
    value="Java"
  >
    <span
      data-v-b209dafd=""
    >
      Java
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(243, 75, 125); --hover-bg-color: #f34b7d; --hover-text-color: #111111;"
    value="C++"
  >
    <span
      data-v-b209dafd=""
    >
      C++
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #ffffff; --hover-text-color: #111111;"
    value="Typescript"
  >
    <span
      data-v-b209dafd=""
    >
      Typescript
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  
  <button
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #222222;"
  >
    <span
      data-v-b209dafd=""
    >
       Clear Filter 
    </span>
  </button>
</div>
`;

exports[`<ReposFilter /> > should have more than one button 1`] = `
<div
  class="w-full grid grid-cols-2 gap-2 md:flex md:justify-around md:flex-wrap p-2"
  data-v-b209dafd=""
>
  
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #ffffff; --hover-text-color: #111111;"
    value="Javascript"
  >
    <span
      data-v-b209dafd=""
    >
      Javascript
    </span>
    <span
      data-v-b209dafd=""
    >
      3
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(227, 76, 38); --hover-bg-color: #e34c26; --hover-text-color: #111111;"
    value="HTML"
  >
    <span
      data-v-b209dafd=""
    >
      HTML
    </span>
    <span
      data-v-b209dafd=""
    >
      2
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(53, 114, 165); --hover-bg-color: #3572A5; --hover-text-color: #111111;"
    value="Python"
  >
    <span
      data-v-b209dafd=""
    >
      Python
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(204, 52, 45); --hover-bg-color: #CC342D; --hover-text-color: #111111;"
    value="Ruby"
  >
    <span
      data-v-b209dafd=""
    >
      Ruby
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(176, 114, 25); --hover-bg-color: #b07219; --hover-text-color: #111111;"
    value="Java"
  >
    <span
      data-v-b209dafd=""
    >
      Java
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(243, 75, 125); --hover-bg-color: #f34b7d; --hover-text-color: #111111;"
    value="C++"
  >
    <span
      data-v-b209dafd=""
    >
      C++
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #ffffff; --hover-text-color: #111111;"
    value="Typescript"
  >
    <span
      data-v-b209dafd=""
    >
      Typescript
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  
  <button
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #222222;"
  >
    <span
      data-v-b209dafd=""
    >
       Clear Filter 
    </span>
  </button>
</div>
`;

exports[`<ReposFilter /> > should have only one button 1`] = `
<div
  class="w-full grid grid-cols-2 gap-2 md:flex md:justify-around md:flex-wrap p-2"
  data-v-b209dafd=""
>
  
  
  <button
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #222222;"
  >
    <span
      data-v-b209dafd=""
    >
       Clear Filter 
    </span>
  </button>
</div>
`;

exports[`<ReposFilter /> > should init without class selected 1`] = `
<div
  class="w-full grid grid-cols-2 gap-2 md:flex md:justify-around md:flex-wrap p-2"
  data-v-b209dafd=""
>
  
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #ffffff; --hover-text-color: #111111;"
    value="Javascript"
  >
    <span
      data-v-b209dafd=""
    >
      Javascript
    </span>
    <span
      data-v-b209dafd=""
    >
      3
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(227, 76, 38); --hover-bg-color: #e34c26; --hover-text-color: #111111;"
    value="HTML"
  >
    <span
      data-v-b209dafd=""
    >
      HTML
    </span>
    <span
      data-v-b209dafd=""
    >
      2
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(53, 114, 165); --hover-bg-color: #3572A5; --hover-text-color: #111111;"
    value="Python"
  >
    <span
      data-v-b209dafd=""
    >
      Python
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(204, 52, 45); --hover-bg-color: #CC342D; --hover-text-color: #111111;"
    value="Ruby"
  >
    <span
      data-v-b209dafd=""
    >
      Ruby
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(176, 114, 25); --hover-bg-color: #b07219; --hover-text-color: #111111;"
    value="Java"
  >
    <span
      data-v-b209dafd=""
    >
      Java
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(243, 75, 125); --hover-bg-color: #f34b7d; --hover-text-color: #111111;"
    value="C++"
  >
    <span
      data-v-b209dafd=""
    >
      C++
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  <button
    class=""
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #ffffff; --hover-text-color: #111111;"
    value="Typescript"
  >
    <span
      data-v-b209dafd=""
    >
      Typescript
    </span>
    <span
      data-v-b209dafd=""
    >
      1
    </span>
  </button>
  
  <button
    data-v-b209dafd=""
    style="color: rgb(255, 255, 255); --hover-bg-color: #222222;"
  >
    <span
      data-v-b209dafd=""
    >
       Clear Filter 
    </span>
  </button>
</div>
`;
